cardano-cli/src/Cardano/CLI/Orphans.hs

```diff
@@ -23,3 +24,6 @@ instance (L.EraTxOut ledgerera, L.EraGov ledgerera) => ToJSON (L.NewEpochState l
       , "rewardUpdate" .= nesRu
       , "currentStakeDistribution" .= nesPd
       ]
+
+instance ToJSON HashableScriptData where
+  toJSON = scriptDataToJsonDetailedSchema
```

Comment:
This is an important choice to review.

Reply:
Vide comment from `cardano-api`:
``` haskell
-- | Warning: Creating 'HashableScriptData' from a 'ScriptData' value pretty
-- much guarantees the original bytes used to create the 'ScriptData'
-- value will be different if we serialize `HashableScriptData` again.
-- Do not use this.
unsafeHashableScriptData :: ScriptData -> HashableScriptData
unsafeHashableScriptData sd = HashableScriptData (serialiseToCBOR sd) sd
```
I understand that it means that we can't be sure that deserialisation from JSON representation into `HashableScriptData` will round trip. 

If that is the case here really, we should prohibit from deserialising from JSON I think, for example like that:
```haskell
instance TypeError (Text "Cannot deserialise HashableScriptData." :$$:
                    Text "Some good justification")
      => FromJSON HashableScriptData where
fromJSON = error "unreachable"
```

Reply:
> I understand that it means that we can't be sure that deserialisation from JSON representation into HashableScriptData will generate the value with the same CBOR representation. 

Ledger's CBOR serialization is not guaranteed to round trip and this cab change the hash. See https://github.com/IntersectMBO/cardano-ledger/issues/2943#issuecomment-1203989504

However I don't see the issue here. `scriptDataToJsonDetailedSchema` does not call `unsafeHashableScriptData`. If we were rendering the hash (which we should) then we have to be careful and use `hashScriptDataBytes`.

Reply:
>However I don't see the issue here. scriptDataToJsonDetailedSchema does not call unsafeHashableScriptData. If we were rendering the hash (which we should) then we have to be careful and use hashScriptDataBytes.
Yes, but it does **exactly the same** - it actually should call it.

Reply:
@Jimbo4350 there's no issue. It's about making hidden assumptions clear. If someone writes naive `FromJSON` instance those two expressions could give different results:
```haskell
hsd = undefined :: HashableScriptData

Right $ hashScriptDataBytes hsd
```
and 
```haskell
scriptDataFromJsonNoSchema <$> (Aeson.eitherDecode $ Aeson.encode hsd)
```

My point is to make that non-roundtripping assumption clearer, or somehow expressed in the type system.